### PR TITLE
feat: integrate stripe

### LINKED ISSUE #296: Integrate Stripe
Take credit card details from the user and send the token to the server
working on it

### PR DESCRIPTION
**Pull request by nikit19**
fixes #296 
gif:
ezgif com-video-to-gif 1

### CODE REVIEW

File: app/src/main/res/layout/fragment_attendee.xml [hunk 1]
```diff
@@ -102,6 +102,12 @@
         android:layout_height="wrap_content">
     </android.support.v7.widget.AppCompatSpinner>
 
+            <com.stripe.android.view.CardInputWidget
+                android:id="@+id/cardInputWidget"
+                android:layout_width="match_parent"
+                android:visibility="gone"
+                android:layout_height="wrap_content" />
+
```
Comment (dreadpool2) [inline app/src/main/res/layout/fragment_attendee.xml]: Indentation :+1: 
Good job overall!

**Comment by nikit19** [comment]
@iamareebjamal Please review

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt [hunk 2]
```diff
@@ -36,6 +40,7 @@ class AttendeeFragment : Fragment() {
     private lateinit var eventId: EventId
     private var ticketIdAndQty: List<Pair<Int, Int>>? = null
     private lateinit var selectedPaymentOption: String
+    private val API_KEY = "Enter your Stripe API key"
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: In build.gradle, loaded as environment variable

**Comment by nikit19** [comment]
@iamareebjamal PR updated.

**Comment by nikit19** [comment]
@iamareebjamal Please review

**Comment by nikit19** [comment]
@iamareebjamal This is a blocker for payments related work. We have a meeting this Tuesday it would be nice if we could complete the payments part by then. Could you please review

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt [hunk 3]
```diff
@@ -39,6 +44,8 @@ class AttendeeFragment : Fragment() {
     private lateinit var selectedPaymentOption: String
     private lateinit var paymentCurrency: String
 
+    private val API_KEY = BuildConfig.STRIPE_API_TOKEN ?: "Enter your Stripe API key"
```
[the remark this answers is not in the document]
Comment (nikit19) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: @iamareebjamal This is something that I wanted to confirm.
I think I am not able to set the environment variable properly but I guess the code is correct.
I am using this command to set the environment variable in the terminal
`export STRIPE_API_TOKEN="myToken"
`
and then I run the app.
But the value is always null

[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeFragment.kt]: Set environment variable through Run Configuration, or compile the app using command line

**Comment by nikit19** [comment]
@iamareebjamal On running the app through the terminal the build fails with following error


`/home/nikit/open-event-android-general/app/build/generated/source/buildConfig/debug/org/fossasia/openevent/general/BuildConfig.java:14: error: cannot find symbol
  public static final String STRIPE_API_TOKEN = myApiKey;
                                                ^
  symbol:   variable myApiKey
  location: class BuildConfig
1 error
`

**Comment by iamareebjamal** [comment]
Because you've not set the environment variable within quotes. But that's not the correct way. @batbrain7 please help

**Comment by dreadpool2** [comment]
@iamareebjamal @nikit19 Testes PR locally, working flawlessly :+1:

**Comment by simarsingh24** [comment]
works fine locally :+1

**Comment by iamareebjamal** [comment]
Will merge after setting the Stripe API key

**Comment by nikit19** [comment]
@iamareebjamal Please merge. If it is possible it would be nice if this could get merged before tomorrow's meeting.

**Comment by nikit19** [comment]
@iamareebjamal I have added the test stripe api key as an environment variable. I think now we can merge this PR safely

**Comment by iamareebjamal** [comment]
Follow up with PR which gracefully handles Stripe API key not being present

**Comment by nikit19** [comment]
@iamareebjamal currently if the API key is not present a toast message is shown to the user that the API key is invalid. ( onError method handles it)
I didn't understand what could be a better way.

**Comment by iamareebjamal** [comment]
Is the field still shown? Is there any edge case where error can happen?

**Comment by nikit19** [comment]
@iamareebjamal yes the field is shown. In case of any error appropriate Toast message is displayed.
I think the field should be shown even if the API key is not provided, since it is the development version. Anyone can try the app and see that the API key is not provided and add it, otherwise they won't know about it.

**Pull request merged**